### PR TITLE
Add ability to pass custom options to ktlint.

### PR DESCRIPTION
**Pull request by runningcode**
This adds the ability to pass custom options to ktlint such as
indentation size and max line length.

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
You also need to be able to add this to the `KtlintGradle` step as well.

File: lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java [hunk 1]
```diff
@@ -69,8 +82,10 @@ public static String defaultVersion() {
 		private final boolean isScript;
 		/** The jar that contains the eclipse formatter. */
 		final JarState jarState;
+		private final Map<String, String> userData;
 
-		State(String version, Provisioner provisioner, boolean isScript) throws IOException {
+		State(String version, Provisioner provisioner, boolean isScript, Map<String, String> userData) throws IOException {
```
Comment (nedtwigg) [inline lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java]: When you store the `Map<String, String>`, I would copy it into a `SortedMap<String, String>` first.  Up-to-date checking is not based on `.equals()`, but rather on binary-equals of the serialized representation.  If you make a `Map<String, String>`, the ordering might change depending on the bucket size, even though the data hasn't changed.

File: lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java [hunk 2]
```diff
@@ -54,7 +59,15 @@ private static FormatterStep create(String version, Provisioner provisioner, boo
 		Objects.requireNonNull(version, "version");
```
Comment (nedtwigg) [inline lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java]: You can replace this whole body with just `return create(version, provisioner, isScript, Collections.emptyMap())`.

**Comment by runningcode** [comment]
Ok, made the changes requested and also updated README and CHANGES!

File: lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java [hunk 3]
```diff
@@ -39,28 +39,12 @@ private KtLintStep() {}
 	static final String NAME = "ktlint";
 	static final String MAVEN_COORDINATE = "com.github.shyiko:ktlint:";
 
-	public static FormatterStep create(Provisioner provisioner) {
-		return create(defaultVersion(), provisioner);
-	}
-
-	public static FormatterStep create(String version, Provisioner provisioner) {
-		return create(version, provisioner, false);
-	}
```
Comment (nedtwigg) [inline lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java]: Deleting all these public static methods breaks the API.  Bring them back :)

**Comment by runningcode** [comment]
Updated description. Ready to merge unless there's anything else we'd like from this! :)

File: lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java [hunk 4]
```diff
@@ -43,18 +44,22 @@ public static FormatterStep create(Provisioner provisioner) {
...
 
 	public static FormatterStep create(String version, Provisioner provisioner) {
-		return create(version, provisioner, false);
+		return create(version, provisioner, Collections.emptyMap());
 	}
 
-	public static FormatterStep createForScript(String version, Provisioner provisioner) {
-		return create(version, provisioner, true);
+	public static FormatterStep create(String version, Provisioner provisioner, Map<String, String> userData) {
+		return create(version, provisioner, false, userData);
```
Comment (nedtwigg) [inline lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java]: `createForScript(String version, Provisioner provisioner)` is still missing.

Comment (runningcode) [reply lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java]: oops. added it back!

**Comment by runningcode** [comment]
@nedtwigg ping! any other changes requested for this PR?

**Comment by nedtwigg** [comment]
Sorry for delay, I had one last comment to give, but it was easier to just make the change.  Thanks for the PR!

**Pull request merged**